### PR TITLE
Added new texture driver related functionality

### PR DESCRIPTION
**Pull request by ansurya**
JIRA Issues: SWDEV-198206

### CODE REVIEW

File: include/hip/hcc_detail/driver_types.h [hunk 1]
```diff
@@ -333,4 +333,22 @@ typedef enum hipFunction_attribute {
...
+    HIP_TR_FILTER_MODE_POINT  = 0, /**< Point filter mode */
+    HIP_TR_FILTER_MODE_LINEAR = 1  /**< Linear filter mode */
+} hipFilter_mode;
+
+typedef enum hipResourcetype_enum {
+    HIP_RESOURCE_TYPE_ARRAY           = 0x00, /**< Array resoure */
+    HIP_RESOURCE_TYPE_MIPMAPPED_ARRAY = 0x01, /**< Mipmapped array resource */
+    HIP_RESOURCE_TYPE_LINEAR          = 0x02, /**< Linear resource */
+    HIP_RESOURCE_TYPE_PITCH2D         = 0x03  /**< Pitch 2D resource */
+} hipResourcetype;
```
Comment (emankov) [inline include/hip/hcc_detail/driver_types.h]: 1. In HIP we decided to use the same name for enum, and for its typedef alias; please see the above `hipFunction_attribute`.
2. I'd prefer seeing these three enums rather in **CUDA Runtime API notation**, cause they exist in both APIs: in Driver and Runtime: `CUaddress_mode `= `cudaTextureAddressMode`, `CUfilter_mode` = `cudaTextureFilterMode`, etc.

File: include/hip/hcc_detail/hip_runtime_api.h [hunk 2]
```diff
@@ -3379,10 +3379,14 @@ hipError_t hipTexRefGetArray(hipArray_t* array, textureReference tex);
 
 hipError_t hipTexRefSetAddressMode(textureReference* tex, int dim, hipTextureAddressMode am);
 
+hipError_t hipTexRefSetAddressMode(textureReference* tex, int dim, hipAddress_mode am);
+
 hipError_t hipTexRefGetAddressMode(hipTextureAddressMode* am, textureReference tex, int dim);
 
 hipError_t hipTexRefSetFilterMode(textureReference* tex, hipTextureFilterMode fm);
 
+hipError_t hipTexRefSetFilterMode(textureReference* tex, hipFilter_mode fm);
```
Comment (emankov) [inline include/hip/hcc_detail/hip_runtime_api.h]: I believe, it is a single function.

**Comment by gargrahul** [comment]
@emankov I have added hold_off_merge label to this PR. I have been working on related changes too. I will submit a new PR hopefully by tomorrow. Let us review then.

**Comment by emankov** [comment]
@gargrahul, @bensander, @mangupta,
Since when we have **race conditions on new features** to implement? Why we are doing almost the same simultaneously and reviewing twice?

**Comment by emankov** [comment]
@gargrahul, FMPOV, the below is not a fair play.
>I have added hold_off_merge label to this PR. I have been working on related changes too. I will submit a new PR hopefully by tomorrow.

I believe that first come, first served.

**Comment by mangupta** [comment]
Closing all existing pull requests that do not target the develop branch. In case this pull request is still valid, please rebase your changes against develop branch and resubmit.

**Pull request closed without merging**